mech/__main__.py

```diff
@@ -242,11 +332,14 @@ def main(args=None):
 
     elif arguments['ssh']:
         name = arguments.get("--user")
-        if not name:
-            name = "mech"
         mech_ssh(name)
         exit()
 
+    elif arguments['scp']:
+        name = arguments.get("--user")
```

Comment:
if `name = None`, then it defaults to the default username put in the `mechfile` which by default is `"mech"`